### DISCUSSION TITLE
MET

### DISCUSSION THREAD

**Discussion by Banias1305** [dtcenter/METplus#2046] [File I/O] [answered]
HELLO AGAIN,

ITS MY FIRST TIME USING MET V10.1.1 . I WANT TO DO AN EVALUATION WRF VS OBS (t2m, wind,mslp, prec). CAN SOMEONE PLEASE HELP ME. I THINK I MAYBE MADE THE CONGI_FILES FOR T2M AND MSLP , BUT WHEN I TRY TO RUN met :
DEBUG 1: Default Config File: /opt/MET_10.1.1/share/met/config/PointStatConfig_default
DEBUG 1: User Config File: /home/st13/TEMPSF/confi
ERROR  :
ERROR  : PointStatConfInfo::process_config() -> "message_type_group_map" must contain an entry for "SURFACE".
ERROR  :

and the confi_file is the following:
model = "WRF";
desc = "NA";
regrid = {
   to_grid    = NONE;
   method     = NEAREST;
   width      = 1;
   vld_thresh = 0.5;
   shape      = SQUARE;}
censor_thresh  = [];
censor_val     = [];
mpr_column     = [];
mpr_thresh     = [];
cQat_thresh     = [ NA ];
cnt_thresh     = [ NA ];
cnt_logic      = UNION;
wind_thresh    = [ NA ];
wind_logic     = UNION;
eclv_points    = 0.05;
hss_ec_value   = NA;
rank_corr_flag = FALSE;
fcst = {
   field = [ { name       = "TMP";
               level      = [ "Z002" ];
               cat_thresh = [ > 273 ];} ]; }
obs = fcst;
message_type   = [ "ADPSFC" ];
sid_inc        = ["Station_ID"];
sid_exc        = [];
obs_quality_inc  = [];
obs_quality_exc  = [];
duplicate_flag = NONE;
obs_summary    = NONE;
obs_perc_value = 50;
station_id = [];
message_type_group_map = [
   { key = "ONLYSF"; val = "ADPSFC"; } ];
//
// Climatology data
///
climo_mean = {

   file_name = [];
   field     = [];

   regrid = {
      method     = NEAREST;
      width      = 1;
      vld_thresh = 0.5;
      shape      = SQUARE;
   }

   time_interp_method = DW_MEAN;
   day_interval       = 31;
   hour_interval      = 6;
}

climo_stdev = climo_mean;
climo_stdev = {
   file_name = [];
}
climo_cdf = {
   cdf_bins    = 1;
   center_bins = FALSE;
   write_bins  = TRUE;
   direct_prob = FALSE;
}
land_mask = {
   flag      = FALSE;
   file_name = [];
   field     = { name = "LAND"; level = "L0"; }
   regrid    = { method = NEAREST; width = 1; }
   thresh    = eq1;}
topo_mask = {
   flag               = FALSE;
   file_name          = [];
   field              = { name = "TOPO"; level = "L0"; }
   regrid             = { method = BILIN; width = 2; }
   use_obs_thresh     = ge-100&&le100;
   interp_fcst_thresh = ge-50&&le50;}
obs_window = {
   beg = -5400;
   end =  5400;}
mask = {
   grid  = [ "FULL" ];
   poly  = [];
   sid   = ["Station_ID"];
   llpnt = [];}
ci_alpha  = [ 0.05 ];
boot = {
   interval = PCTILE;
   rep_prop = 1.0;
   n_rep    = 0;
   rng      = "mt19937";
   seed     = "";}

interp = {
   vld_thresh = 1.0;
   shape      = SQUARE;

   type = [
      {
         method = NEAREST;
         width  = 1;
      }
   ];
}
hira = {
   flag            = FALSE;
   width           = [ 2, 3, 4, 5 ];
   vld_thresh      = 1.0;
   cov_thresh      = [ ==0.25 ];
   shape           = SQUARE;
   prob_cat_thresh = [];}
output_flag = {
   fho    = NONE;
   ctc    = NONE;
   cts    = BOTH;
   mctc   = NONE;
   mcts   = NONE;
   cnt    = NONE;
   sl1l2  = NONE;
   sal1l2 = NONE;
   vl1l2  = NONE;
   val1l2 = NONE;
   vcnt   = NONE;
   pct    = NONE;
   pstd   = NONE;
   pjc    = NONE;
   prc    = NONE;
   ecnt   = NONE; // Only for HiRA.
   orank  = NONE; // Only for HiRA.
   rps    = NONE; // Only for HiRA.
   eclv   = NONE;
   mpr    = NONE;}
tmp_dir        = "/tmp";
output_prefix  = "MSLP_OUT";
version        = "V10.1.0";

**Comment by jprestop** [comment]
Hi @Banias1305.  Thank you for your question.  Please take a look at the documentation for message_type_group_map [here](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html?highlight=message_type_group_map#message-type-group-map) and try setting message_type_group_map in the following way:
```
mesage_type_group_map = [
   { key = "SURFACE"; val = "ADPSFC,SFCSHP,MSONET";               },
   { key = "ANYAIR";  val = "AIRCAR,AIRCFT";                      },
   { key = "ANYSFC";  val = "ADPSFC,SFCSHP,ADPUPA,PROFLR,MSONET"; },
   { key = "ONLYSF";  val = "ADPSFC,SFCSHP";                      }
];
```
modifying the "val" settings from there.  The usage of "SURFACE" is required, hence the error you are receiving. The land and water entries are used when land_mask.flag = true.  Please let us know how it goes.

**Reply by jprestop** [reply]
Hi @Banias1305.  I just wanted to follow up on this Discussion and see if you were able to resolve the problem you were experiencing?  Do you have any remaining questions?

**Comment by Banias1305** [comment]
Hi,       Sorry, I haven't sent further information on this, but I'm trying to run the point_stat command for each month for each Var and it's taking me time. -I am sending you the script I created to run the point_stat command for each month:#Define the range of time stepsfor i in $(seq -f "%03g" 6 3 96)do  # Create a directory for each time step  mkdir -p POINT_STAT/12/$i
  # Loop through each day of January  for d in $(seq -f "%02g" 1 31)  do    # Skip days that don't exist in January    if [ $d -gt 31 ]    then      continue    fi
    # Run the point_stat command and redirect output to the corresponding directory    point_stat /home/st13/WRF_M/12/${d}/WRFPRS_d02.$i /home/st13/MSLP/prmsl1.nc /home/st13/FILE_CONG/confi  -outdir POINT_STAT/12/$i             #  in the confi_file i choose CNT & MPR as output   donedone- I then run the following script with the same logic for stat_analysis by FCST_VAR,OBS_SID to get my final stats: [I get for each FCST_LEAD for each MONTH]
for m in $(seq -f "%02g" 1 12); do for i in $(seq -f "%03g" 6 96); do    # Create a directory for each time step    mkdir -p STAT/$m/$i    # Run stat_analysis on all mpr files for the current time step   f="/home/st13/MSLP/POINT_STAT/$m/$i/*_mpr.txt"    #f="/home/st13/MSLP/POINT_STAT/01/006/*_mpr.txt" (EXAMPLE)     stat_analysis -lookin $f -job aggregate_stat -line_type MPR -out_line_type CNT -by FCST,OBS -out STAT/$m/$i/out.txt  donedone I would like your opinion on the above scripts and especially on stat_analysis .



    On Tuesday, March 7, 2023 at 05:57:11 PM GMT+2, jprestop ***@***.***> wrote:  
 
 


Hi @Banias1305. I just wanted to follow up on this Discussion and see if you were able to resolve the problem you were experiencing? Do you have any remaining questions?

**Reply by jprestop** [reply]
HI @Banias1305.  

> I would like your opinion on the above scripts and especially on stat_analysis.

Could you please provide more information about what information you'd like us to provide you with, with regard to your script?

**Comment by Banias1305** [comment]
In the above scripts I choose to run every three hours and just in the confi_stat default file I only declare the variable I want (e.g. TMP/Z2) and don't change anything else. I do not configure the files via pcpcombine tool. Is this step considered correct?Is there a way to declare in the confi_file (point_stat) that my observations are three-hourly?          Also, from running the point stat command I get 3 kinds of output cnt.txt/mpr.txt/.stat . From these I choose to use mpr.txt in stat_analysis and then output cnt stats. 
I think I will need the pcpcombine tool only in case of rainfall, right? Reformatting the FCST data to 6-hourly as my OBSERVATIONS As you can see my questions are purely about the operation of MET and its corresponding tools, because as I've told you it's the first time I'm using it for my thesis. I am trying on my own to understand how each tool runs and results each of them gives me.

Στάλθηκε από το Ταχυδρομείο Yahoo σε Android 
 
  Στις Τρί, 7 Μαρ, 2023 στις 21:24, ο ***@***.***> έγραψε:   


HI @Banias1305.


I would like your opinion on the above scripts and especially on stat_analysis.


Could you please provide more information about what information you'd like us to provide you with, with regard to your script?

**Reply by jprestop** [reply]
Hi @Banias1305.  I don't have the expertise to answer these questions.  @JohnHalleyGotway could you please take a look at @Banias1305's questions and provide feedback.

**Reply by JohnHalleyGotway** [reply, marked as answer]
@Banias1305, there are several options for how you might call the MET tools to achieve the type of analysis you want.

It sounds like right now, you're running Point-Stat to generate matched pair output (MPR line type). And then you're reading those MPR lines into Stat-Analysis and computing continuous statistics. That approach is useful if you want to compute stats separately for each station, but it looks like you are NOT doing that in your Stat-Analysis job.

Let me answer a couple of the questions you raised:

1. Yes, you only need to run pcp_combine to make your forecast and/or observation precipitation accumulation intervals consistent. You do not, in general, need to run it for other variables, like temperature. However, the `-derive` option in pcp_combine is useful if you want to compute a daily forecast maximum or minimum temperature.
2. The Point-Stat tool searches for point observations using a matching time window that is defined relative the current valid time being evaluated. That's defined in the config file by the `obs_window` dictionary. Here's the [documentation](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html?highlight=obs_window#obs-window) about that.

Stepping back, here's a couple options for how you might configure these tools. Up to you to decide what you'd really like to do:

1. Use Point-Stat and Stat-Analysis to compute stats separately for each station:
  - Run Point-Stat to write out the MPR line type.
  - Run Stat-Analysis to aggregate those MPR lines separately for each station:
`-job aggregate_stat -line_type MPR -out_line_type CNT -by OBS_SID,FCST_VAR,FCST_LEV -out_stat MPR_to_CNT.stat`

2. Use Point-Stat and Stat-Analysis to compute stats aggregated over spatial regions:
  - Define masking regions as subsets of your domain over which you want aggregate statistics. Use `mask.grid = "FULL"` to just use all points in the forecast grid. Or run the Gen-Vx-Mask tool to define more complex masking regions.
  - Run Point-Stat to write out both the SL1L2 and CNT line types. The CNT line contains stats for that individual run.
  - Run Stat-Analysis to aggregate the SL1L2 lines across multiple runs:
 `-job aggregate_stat -line_type SL1L2 -out_line_type CNT -by FCST_VAR,FCST_LEV -out_stat SL1L2_to_CNT.stat`

The MET tools were designed to be flexible in how they are configured and the order in which you run them. I realize that's challenging when you're looking for the "right answer". But it really just depends exactly what you're trying to accomplish.

That's one reason to use METplus INSTEAD OF writing your own custom scripts. The existing [METplus Use Cases ](https://metplus.readthedocs.io/en/latest/Users_Guide/usecases.html)provide good examples that are guaranteed to be reasonable ways of configuring/running the MET tools.

**Reply by Banias1305** [reply]
Hi @JohnHalleyGotway ,
1) I followed your advice and ran the command :job aggregate_stat -line_type MPR -out_line_type CNT -by OBS_SID,FCST_VAR,FCST_LEV -out_stat MPR_to_CNT.stat 
for PRMSL/Z0, RH/Z2, TMP/Z2.  I like this suggestion , I think it is better to create an out_stat for each St_ID. I am grateful for this, thank you.
2)  pcp_combine -sum 20210101_06 6 20210102_00 24 sample_6.nc -pcpdir WRFPRS_d02.* -pcprx WRFPRS_d02.*
I want to write a single NetCDF file containing this 6-hour fcst accumulation data and then through point_stat to evaluate it with the 6-hour OBS data I have. I put -pcprx because the FCST data format is WRFPRS_d02.{006-096}/per day . What am I doing wrong in the above command? I think [- add], I can't use it because of the format of my files.

**Reply by JohnHalleyGotway** [reply]
@Banias1305 just following up on old discussions. Do any more questions remain on this or can we mark it as being "answered"?
